### DISCUSSION TITLE
"(outline) No response from provider when requesting symbols!"

### DISCUSSION THREAD

**Discussion by MoltenMonster** [hedyhli/outline.nvim#73] [Help]
When I do `:OutlineOpen` on a markdown file it says `(outline) No response from provider when requesting symbols!`, when I do so on a Lua file its blank. I have LSPs installed for both. I don't have any configuration other than `require("outline").setup({})` in my init.lua, am I supposed to? the repo page shows the defaults and LSP is on there.

**Comment by hedyhli** [comment]
What LSP are you using for markdown? You can find out which one outline.nvim is calling using `:OutlineStatus`.

As for lua, this shouldn't be happening and I certainly am not able to reproduce it. Could you show or link to your entire config, including where this plugin is declared in your plugin manager?

**Reply by MoltenMonster** [reply]
I got it working on Markdown now, but it's blank on this lua script:

```lua
local TargetPartName = "HumanoidRootPart"
local GenericHumanoidTargetPartName = TargetPartName	
local LineColor = Color3.new(255, 255, 255)
local TeammateLineColor = Color3.new(0, 0.25, 1)
local GenericHumanoidLineColor = Color3.new(1, 0, 0)
local LineWidth = 2
local DrawTeammates = true
local FindHumanoids = true
local GetLineOrigin = function(Camera)
	return Vector2.new(Camera.ViewportSize.X/2, Camera.ViewportSize.Y*.9)
end
  
local Camera = workspace.CurrentCamera
local LineOrigin = nil
Camera:GetPropertyChangedSignal("ViewportSize"):Connect(function()
	LineOrigin = GetLineOrigin(Camera)
end)
LineOrigin = GetLineOrigin(Camera)

local Gui = Instance.new("ScreenGui")
Gui.Name = "SnaplineGui"
Gui.Parent = game.Players.LocalPlayer.PlayerGui
local Lines = {}
function Setline(Line, Width, Color, Origin, Destination)
	local Position = (Origin + Destination) / 2
	Line.Position = UDim2.new(0, Position.X, 0, Position.Y)
	local Length = (Origin - Destination).Magnitude
	Line.BackgroundColor3 = Color
	Line.BorderColor3 = Color
	Line.Size = UDim2.new(0, Length, 0, Width)
	Line.Rotation = math.deg(math.atan2(Destination.Y - Origin.Y, Destination.X - Origin.X))
end


game:GetService("RunService").RenderStepped:Connect(function()
	local Targets = {}
	for i, Player in pairs(game.Players:GetPlayers()) do
		if Player == game.Players.LocalPlayer then continue end
		local IsTeammate = game.Players.LocalPlayer.Team ~= nil and Player.Team == game.Players.LocalPlayer.Team
		if not DrawTeammates and IsTeammate then continue end
		local Character = Player.Character
		if not Character then continue end
		local TargetPart = Character:FindFirstChild(TargetPartName)
		if not TargetPart then continue end
		local ScreenPoint, OnScreen = Camera:WorldToScreenPoint(TargetPart.Position)
		if OnScreen then
			table.insert(Targets, {Vector2.new(ScreenPoint.X, ScreenPoint.Y), IsTeammate and TeammateLineColor or LineColor})
		end
	end
	if FindHumanoids then
		for _, Obj in pairs(workspace:GetDescendants()) do
			if Obj.ClassName ~= "Humanoid" then continue end
			if game.Players:FindFirstChild(Obj.Parent.Name) then continue end
			local TargetPart = Obj.Parent:FindFirstChild(TargetPartName)
			if not TargetPart then continue end
			local ScreenPoint, OnScreen = Camera:WorldToScreenPoint(TargetPart.Position)
			if OnScreen then
				table.insert(Targets, {Vector2.new(ScreenPoint.X, ScreenPoint.Y), GenericHumanoidLineColor})
			end
		end
	end
	if #Targets > #Lines then
		local NewLine = Instance.new("Frame")
		NewLine.Name = "Snapline"
		NewLine.AnchorPoint = Vector2.new(.5, .5)
		NewLine.Parent = Gui
		table.insert(Lines, NewLine)
	end
	for i, Line in pairs(Lines) do
		local TargetData = Targets[i]
		if not TargetData then
			Line:Destroy()
			table.remove(Lines, i)
			continue
		end
		Setline(Line, LineWidth, TargetData[2], LineOrigin, TargetData[1])
	end
end)
```

**Comment by MoltenMonster** [comment]
@hedyhli btw can I search the outline for a phrase and have it show results?

**Reply by hedyhli** [reply]
For now, you can absolutely use `/<search-term>`, select the symbol, then `:noh`.

You can even set up an autocmd that will clear the search whenever cursor leaves an outline buffer. With regards to having outline.nvim support this built-in (or with a config option) can be explored in #46 where I'm happy to discuss and review PRs.

Something that will visually "filter-out" symbols that do not match the search has been requested before (#31) but deemed out-of-scope as the implementation might become a little involved.

**Comment by iamxiaojianzheng** [comment]
This error is displayed when I try to trigger Outline when the lsp is not fully loaded.

I can understand this behavior, but when I trigger the Outline again after the lsp is loaded, it still returns an error.

And when I open other files of the same type, I still get an error.

This does not occur when the Outline is triggered only after the vim is restarted and the lsp is completely loaded.

In my memory, there was a time when this was not the case.

When I trigger the Outline before the lsp is loaded, it does not report an error immediately. Instead, it waits until the lsp is loaded and then the outline window pops up.

I don't know why it can't be like this later.

**Reply by hedyhli** [reply]
Hi, I can't reproduce this. When I open a file and immediately open outline, it waits until LSP is ready until the sidebar is opened.

> In my memory, there was a time when this was not the case.

Perhaps it's another plugin that is interfering? Or have you recently updated a config value in outline.nvim?
